File: swift/validAnagram.swift
func isAnagram(s: String, _ t: String) -> Bool {
    let sArray = Array(s.characters)
    let tArray = Array(t.characters)
    var sMap = [Character: Int]()
    var tMap = [Character: Int]()
    for c in sArray {
        if let count = sMap[c] {
            sMap[c] = count + 1
        } else {
            sMap[c] = 1
        }
    }
    for c in tArray {
        if let count = tMap[c] {
            tMap[c] = count + 1
        } else {
            tMap[c] = 1
        }
    }
    for c in sArray {
        if sMap[c] != tMap[c] {
            return false
        }
    }
    for c in tArray {
        if sMap[c] != tMap[c] {
            return false
        }
    }
    return true
}

File: ruby/valid_anagram.rb
def is_anagram(s, t)
  s_map = {}
  t_map = {}
  s.split("").each do |c|
    if s_map[c] != nil
      s_map[c] += 1
    else
      s_map[c] = 1
    end
  end
  t.split("").each do |c|
    if t_map[c] != nil
      t_map[c] += 1
    else
      t_map[c] = 1
    end
  end
  s.split("").each do |c|
    if s_map[c] != t_map[c]
      return false
    end
  end
  t.split("").each do |c|
    if s_map[c] != t_map[c]
      return false
    end
  end
  return true
end
